### PR TITLE
Add trino-server

### PR DESCRIPTION
**Pull request by jonashaag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/trino-server/build.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Don't build docs (requires Docker).
+sed -E -i'' -e 's@.+<module>docs</module>@@g' pom.xml
+
+# Parallelize the build on Linux.
+# Disabled on non-Linux because of https://issues.apache.org/jira/browse/MNG-7868.
+if [[ "$build_platform" =~ linux ]]; then
```
Comment (xhochy) [inline recipes/trino-server/build.sh:10]: Style nit
```suggestion
if [[ "$build_platform" == linux-* ]]; then
```

File: recipes/trino-server/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{% set name = "trino-server" %}
+{% set version = "433" %}
+{% set java_version = 17 %}
```
Comment (xhochy) [inline recipes/trino-server/meta.yaml:3]: Please either write this without Jinja (preferred) or put the value in a `conda_build_config.yaml`. Avoid top-level Jinja in `meta.yml` recipes as much as possible.

**Review by xhochy — APPROVED** [review]
Looks good, just minor style comments.

**Comment by jonashaag** [comment]
👇
▶️

**Pull request merged**